### PR TITLE
Switch CI to temurin Java distribution, enable Gradle cache via setup-java

### PR DESCRIPTION
**Pull request by Maxr1998**

### CODE REVIEW

File: .github/workflows/extension-publish.yaml [hunk 1]
```diff
@@ -20,17 +20,9 @@ jobs:
       - name: Setup Java
         uses: actions/setup-java@v2
         with:
-          distribution: adopt
+          distribution: temurin
           java-version: 11
```
Comment (nielsvanvelzen) [inline .github/workflows/extension-publish.yaml:24]: Do we want to switch to Java 17 (LTS)?

Comment (Maxr1998) [reply .github/workflows/extension-publish.yaml:24]: No, since we use Kapt for Room which breaks on Java 17.

Comment (nielsvanvelzen) [reply .github/workflows/extension-publish.yaml:24]: That only applies to jellyfin-android right?

Comment (Maxr1998) [reply .github/workflows/extension-publish.yaml:24]: Right now, yes.

**Review by nielsvanvelzen — APPROVED** [review]

**Pull request merged**